src/org/jgroups/protocols/raft/Settable.java

```diff
@@ -38,7 +40,82 @@ default byte[] set(byte[] buf, int offset, int length) throws Exception {
      */
     default byte[] set(byte[] buf, int offset, int length, long timeout, TimeUnit unit) throws Exception {
         CompletableFuture<byte[]> future=setAsync(buf, offset, length);
-        return future.get(timeout, unit);
+        return progressiveWait(future, timeout, unit);
+    }
+
+    int MAX_SPINS = (Runtime.getRuntime().availableProcessors() < 2) ? 0 : 32;
+
+    static byte[] progressiveWait(CompletableFuture<byte[]> future) throws Exception {
+        byte[] result = future.getNow(null);
+        if (result != null) {
+            return result;
+        }
+        if (Thread.currentThread().isInterrupted()) {
+            throw new InterruptedException();
+        }
+        int maxSpins = MAX_SPINS;
+        int maxYield = MAX_SPINS == 0 ? 0 : maxSpins + 20;
+        int maxTinyPark = MAX_SPINS == 0 ? 0 : maxYield + 900;
```

Comment:
These instead of magic numbers based on the typical latencies on linux syscall(s) (for yield and the minimum park time) for recent kernels with Specter/Meltdown mitigation applied 
(see old but good http://blog.vanillajava.blog/2012/04/yield-sleep0-wait01-and-parknanos1.html)